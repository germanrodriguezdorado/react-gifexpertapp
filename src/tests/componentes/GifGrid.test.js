import { shallow } from "enzyme";
import GifGrid from "../../componentes/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe('Funciones de test que prueban el componente GifGrid', () => {
    
    const category = "Spiderman";
    //const wrapper = shallow(<GifGrid category={ category } />);

    test('El componente debe haber sido renderizado correctamente.', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id: "1",
            url: "http://localhost/cualquier/cosa.jpg",
            title: "Cualquier cosa"
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    });
    
    
});
