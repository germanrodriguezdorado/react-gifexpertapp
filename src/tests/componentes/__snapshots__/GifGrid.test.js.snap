// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Funciones de test que prueban el componente GifGrid Debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Spiderman
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Funciones de test que prueban el componente GifGrid El componente debe haber sido renderizado correctamente. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Spiderman
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
